Sources/Core/Sources/GUIState.swift

```diff
@@ -9,6 +9,7 @@ public struct GUIState {
   public var stashedMessageInput: String?
   public var playerMessageHistory: [String] = []
   public var currentMessageIndex: Int?
+  public var messageInputCursor: Int = 0
```

Comment:
Add a doc comment (`/// ...`) to explain that the cursor is indexed backwards to simplify input handling code